src/Nox.Solution/Models/Types/TypeUserInterface/TypeUserInterface.cs

```diff
@@ -2,22 +2,59 @@
 
 namespace Nox.Solution
 {
-
+    [Title("UI display options for this attribute.")]
+    [Description("Specify how this attribute is rendered on the Nox user interface. Configuration options include label, icon, input/output masks, help hints among other.")]
     [AdditionalProperties(false)]
     public class TypeUserInterface : DefinitionBase
     {
+        [Title("A descriptive label for this attribute on the UI.")]
+        [Description("The text label rendered on the user interface alongside the NoxTextInput component.")]
         public string? Label { get; internal set; }
+
+        // Pending implementation details
         public string? TranslationId { get; internal set; }
+
+        // Pending implementation details
         public Widget? Widget { get; internal set; }
+
+        [Title("The icon representing the attribute on the UI.")]
+        [Description("A reference URI to the attribute icon file. Could be a CSS or Mudblazor reference, e.g. '@Icons.Material.Filled.Globe'.")]
         public string? Icon { get; internal set; }
+
+        [Title("The position of the attribute icon on the UI.")]
+        [Description("The postition of the attribute icon relative to the text input area on the user interface, e.g. begin or end.")]
         public IconPosition? IconPosition { get; internal set; } = global::Nox.IconPosition.Begin;
+
+        [Title("A mask expression used to validate and control text input.")]
+        [Description("A string of characters that indicates the format of valid input values. Examples include '*?@?*.?*' for an email address and '(000) 000-0000' for a seven digit phone number with mandatory area code.")]
         public string? InputMask { get; internal set; }
+
+        [Title("A mask expression used to validate and control text output.")]
+        [Description("A string of characters that indicates the format of valid output. Used to handle validation triggers for onSubmit and onChange events of the control.")]
         public string? OutputMask { get; internal set; }
+
+        [Title("The regular expression against which input is validated.")]
+        [Description("A regex string that indicates the format of valid input. Examples include '/^[^@\\s]+@[^@\\s]+\\.[^@\\s]+$/gm' for email address and '/^(\\([0-9]{3}\\) |[0-9]{3}-)[0-9]{3}-[0-9]{4}$/gm' for a seven digit phone number with mandatory area code.")]
```

Comment:
We won't use a mask for email output. numbers, currencies, and password will be masked.

Reply:
Removed email examples for InputMask and Regex.